src/altinn-app-frontend/src/components/message/ErrorReport.tsx

```diff
@@ -1,128 +1,145 @@
 import * as React from 'react';
 
-import { useAppSelector } from 'src/common/hooks';
-import { getUnmappedErrors } from 'src/utils/validation';
-import type { IValidations } from 'src/types';
+import { Panel, PanelVariant } from '@altinn/altinn-design-system';
+import { Grid, makeStyles } from '@material-ui/core';
+
+import { useAppDispatch, useAppSelector } from 'src/common/hooks';
+import { FullWidthWrapper } from 'src/features/form/components/FullWidthWrapper';
+import { renderLayoutComponent } from 'src/features/form/containers/Form';
+import { FormLayoutActions } from 'src/features/form/layout/formLayoutSlice';
+import { getMappedErrors, getUnmappedErrors } from 'src/utils/validation';
+import type { ILayout } from 'src/features/form/layout';
+import type { FlatError } from 'src/utils/validation';
 
 import { getLanguageFromKey } from 'altinn-shared/utils';
 
-const ErrorReport = () => {
-  const validations = useAppSelector(
-    (state) => state.formValidations.validations,
+export interface IErrorReportProps {
+  components: ILayout;
+}
+
+const ArrowForwardIcon =
+  '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="16" style="position: relative; top: 2px">' +
+  '<path d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"></path>' +
+  '</svg>';
+
+const useStyles = makeStyles((theme) => ({
+  errorList: {
+    listStylePosition: 'inside',
+    listStyleImage: `url("data:image/svg+xml,${encodeURIComponent(
+      ArrowForwardIcon,
+    )}")`,
+    '& > li': {
+      marginBottom: theme.spacing(1),
+    },
+  },
+  buttonAsInvisibleLink: {
+    backgroundColor: 'transparent',
+    border: 'none',
+    cursor: 'pointer',
+    textDecoration: 'none',
+    display: 'inline',
+    margin: 0,
+    padding: 0,
+  },
+}));
+
+const ErrorReport = ({ components }: IErrorReportProps) => {
+  const classes = useStyles();
+  const dispatch = useAppDispatch();
+  const currentView = useAppSelector(
+    (state) => state.formLayout.uiConfig.currentView,
   );
-  const unmappedErrors = getUnmappedErrors(validations);
-  const hasUnmappedErrors = unmappedErrors.length > 0;
+  const [errorsMapped, errorsUnmapped] = useAppSelector((state) => [
+    getMappedErrors(state.formValidations.validations),
+    getUnmappedErrors(state.formValidations.validations),
+  ]);
   const language = useAppSelector((state) => state.language.language);
-  const formHasErrors = useAppSelector((state) =>
-    getFormHasErrors(state.formValidations.validations),
-  );
-  const hasSubmitted = useAppSelector((state) => state.formData.hasSubmitted);
-  const errorRef = React.useRef(null);
-
-  React.useEffect(() => {
-    if (hasSubmitted) {
-      errorRef?.current?.focus();
-    }
-  }, [hasSubmitted, unmappedErrors]);
+  const hasErrors = errorsUnmapped.length > 0 || errorsMapped.length > 0;
 
-  if (!formHasErrors) {
+  if (!hasErrors) {
     return null;
   }
 
+  const OnClickError =
```

Comment:
This is handling the event, and thus should be named `handleErrorClick`. It is common to put the "type" (Click) last as well. That way when listing all events, they would be grouped by "name" first, then "type". Hope that makes sense 🙂

```suggestion
  const handleErrorClick =
```